diff --git a/malloc.c b/glibc-2.32/malloc/malloc.c
index 9faa534..ee87ddb 100755
--- a/malloc.c
+++ b/glibc-2.32/malloc/malloc.c
@@ -276,7 +276,7 @@
 */
 
 #ifndef MALLOC_DEBUG
-#define MALLOC_DEBUG 2
+#define MALLOC_DEBUG 0
 #endif
 
 #ifndef NDEBUG
@@ -1779,7 +1779,7 @@ static struct malloc_par mp_ =
   .n_mmaps_max = DEFAULT_MMAP_MAX,
   .mmap_threshold = DEFAULT_MMAP_THRESHOLD,
   .trim_threshold = DEFAULT_TRIM_THRESHOLD,
-#define NARENAS_FROM_NCORES(n) 5
+#define NARENAS_FROM_NCORES(n) ((n) * (sizeof (long) == 4 ? 2 : 8))
   .arena_test = NARENAS_FROM_NCORES (1)
 #if USE_TCACHE
   ,
